main_shared.go

```diff
@@ -181,7 +189,68 @@ func findppid(pid int) int {
 	return 0
 }
 
-func quit() {
-	ztn.UPNPIGDCleanupMapped()
-	os.Exit(0)
+func GenerateCoreDNSConfig(myDNSInfo *godnschange.DNSInfo, domains []string) string {
```

Comment:
Lets move this function and the one below in another file (like main_dns.go or whatever you see fit)